File: editorial/chap4-4/prob4-4-3.cpp
#include <iostream>
using namespace std;

long long N;
long long F[10000009];
long long Answer = 0;

int main() {
	// 入力 → 配列の初期化
	cin >> N;
	for (int i = 1; i <= N; i++) F[i] = 0;
	
	// F[1], F[2], ..., F[N] を計算する
	for (int i = 1; i <= N; i++) {
		// F[i], F[2i], F[3i], ... に 1 を加算
		for (int j = i; j <= N; j += i) F[j] += 1;
	}
	
	// 答えを求める → 出力
	for (int i = 1; i <= N; i++) {
		Answer += 1LL * i * F[i];
	}
	cout << Answer << endl;
	return 0;
}

File: editorial/chap4-4/prob4-4-3.py
# 重要な注意
# Python は遅いので、このプログラムを提出しても、実行時間制限超過（TLE）となります。
# このため、全く同じコードを「PyPy3 (7.3.0)」で提出してください。おそらく AC します。
# 計算量の速い解法を知りたい人は「prob4-4-3_fast.py」をご覧ください。

# 入力
N = int(input())
F = [0 for i in range(N + 1)]

# F[1], F[2], ..., F[N] を計算する
for i in range(1, N + 1):
	# F[i], F[2i], F[3i], ... に 1 を加算
	for j in range(1, (N // i) + 1):
		F[j * i] += 1

# 答えを求める
Answer = 0
for i in range(1, N + 1):
	Answer += i * F[i]

# 出力
print(Answer)